### PR TITLE
Request Detail Page

### PR DESCRIPTION
**Pull request by wolfd**
- Clean up/refactor Request and Task buttons so I can use them.
- So many refactors I don't even remember anymore.
- Remove all that everything.

### CODE REVIEW

**Comment by wolfd** [comment]
@tpetr 💥

**Comment by wolfd** [comment]
@Calvinp @kwm4385 @tpetr 
ready for round 2?

File: SingularityUI/app/components/common/modal/FormModal.jsx [hunk 1]
```diff
@@ -2,10 +2,27 @@ import React from 'react';
...
+import TagsInput from 'react-tagsinput';
+import Duration from '../formItems/Duration';
 
 export default class FormModal extends React.Component {
+  constructor(props) {
+    super(props);
+    const formState = {};
+    _.each(props.formElements, (e) => {
+      formState[e.name] = e.defaultValue && e.defaultValue.toString();
+    });
```
Comment (tpetr) [inline SingularityUI/app/components/common/modal/FormModal.jsx]: why `_.each()` vs. ES6's `.forEach()`?

File: SingularityUI/app/components/common/modal/FormModal.jsx [hunk 2]
```diff
@@ -2,10 +2,27 @@ import React from 'react';
...
+
+    this.state = {
+      visible: false,
+      formState,
+      errors: {}
+    };
+
+    this.hide = this.hide.bind(this);
+    this.show = this.show.bind(this);
+    this.confirm = this.confirm.bind(this);
```
Comment (kwm4385) [inline SingularityUI/app/components/common/modal/FormModal.jsx]: _.bindAll() would be useful here

File: SingularityUI/app/components/common/modal/FormModal.jsx (hunk 1 above)
Comment (wolfd) [inline SingularityUI/app/components/common/modal/FormModal.jsx]: I think the reason this got highlighted is a indentation thing, but I can change that. I prefer the ES6 one, but I think in some of the other PRs we've had usage of underscore's one.

File: SingularityUI/app/components/deployDetail/ActiveTasksTable.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    headers={['Name', 'Last State', 'Started', 'Updated', '', '']}
+    renderTableRow={(data, index) => {
+      return (
+        <tr key={index}>
+          <td><a href={`${config.appRoot}/task/${data.taskId.id}`}>{data.taskId.id}</a></td>
+          <td><span className={`label label-${Utils.getLabelClassFromTaskState(data.lastTaskState)}`}>{Utils.humanizeText(data.lastTaskState)}</span></td>
+          <td>{Utils.timestampFromNow(data.taskId.startedAt)}</td>
+          <td>{Utils.timestampFromNow(data.updatedAt)}</td>
+          <td className="actions-column"><a href={`${config.appRoot}/request/${data.taskId.requestId}/tail/${config.finishedTaskLogPath}?taskIds=${data.taskId.id}`} title="Log">&middot;&middot;&middot;</a></td>
+          <td className="actions-column"><JSONButton object={data}>{'{ }'}</JSONButton></td>
```
Comment (tpetr) [inline SingularityUI/app/components/deployDetail/ActiveTasksTable.jsx]: more of a question than an actual issue / nitpick -- is a JSON button ever anything other than `{ }`? Would it be worth folding it into the element?

Comment (tpetr) [inline SingularityUI/app/components/deployDetail/ActiveTasksTable.jsx]: @kwm4385 pointed out that there are `JSON` buttons at the tops of some pages, so nevermind

File: SingularityUI/app/components/deployDetail/DeployDetail.jsx [hunk 4]
```diff
@@ -199,23 +189,23 @@ class DeployDetail extends React.Component {
...
+          last={true}
           headers={['Task', 'Timestamp', 'Duration', 'Status', 'Message', '']}
           renderTableRow={(data, index) => {
             return (
               <tr key={index}>
                 <td><a href={`${config.appRoot}/task/${data.taskId.id}`}>{data.taskId.id}</a></td>
                 <td>{Utils.absoluteTimestamp(data.timestamp)}</td>
                 <td>{data.durationMillis} {data.durationMillis ? 'ms' : ''}</td>
-                <td>{data.statusCode ? <span className={`label label-${data.statusCode == 200 ? 'success' : 'danger'}`}>HTTP {data.statusCode}</span> : <span className="label label-warning">No Response</span>}</td>
+                <td>{data.statusCode ? <span className={`label label-${data.statusCode === 200 ? 'success' : 'danger'}`}>HTTP {data.statusCode}</span> : <span className="label label-warning">No Response</span>}</td>
```
Comment (tpetr) [inline SingularityUI/app/components/deployDetail/DeployDetail.jsx]: feels a little gross to pack this this logic in here

File: SingularityUI/app/components/deployDetail/ActiveTasksTable.jsx (hunk 3 above)
Comment (wolfd) [inline SingularityUI/app/components/deployDetail/ActiveTasksTable.jsx]: I have something configured other places that uses a default child element to do something like this. But that can be for later.

File: SingularityUI/app/components/deployDetail/DeployDetail.jsx (hunk 4 above)
Comment (wolfd) [inline SingularityUI/app/components/deployDetail/DeployDetail.jsx]: @kwm4385 will get that after this merges

**Comment by tpetr** [comment]
LGTM

**Pull request merged**